File: leetcode/1.two-sum.py
from typing import List
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start



class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}

        for i, num in enumerate(nums):
            substitution = target - num

            if substitution in hashMap:
                return [hashMap[substitution], i]

            hashMap[num] = i


# inputs = [
#     [
#         [2,7,11,15],
#         9
#     ],
#     [
#         [3,2,4],
#         6
#     ],
#     [
#         [3,3],
#         6
#     ]
# ]

# s = Solution()
# for [arr, target] in inputs:
#     print(s.twoSum(arr, target))



# @lc code=end

File: leetcode/1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map()

    for(let i = 0; i < nums.length; i++) {
        const key = nums[i];
        const array = map.get(key);

        if(!array) {
            map.set(key, [i])
        } else {
            array.push(i)
            map.set(key, array)
        }
    }
    
    console.log(map);
};
// @lc code=end

twoSum([1,1,1,2,2], 3)